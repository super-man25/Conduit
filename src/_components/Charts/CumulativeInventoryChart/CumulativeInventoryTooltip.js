// @flow
import React from 'react';
import { ChartTooltip } from '../ChartTooltip';
import { cumulativeTooltip } from '_helpers/chart-utils';

type TooltipContentProps = {
  active?: boolean,
  payload?: any,
  dateFormatter: (Date) => string
};

export function CumulativeInventoryTooltip(props: TooltipContentProps) {
  const { active, dateFormatter, payload } = props;
  if (!active || !payload || !payload.length) {
    return null;
  }
  const [{ payload: stat }] = payload;
  const { inventory, revenue, avgTicketPrice } = cumulativeTooltip(stat);
  const headerText = dateFormatter(stat.timestamp);
  const bodyJson = {
    'Remaining Inventory': inventory,
    'Revenue To Date': revenue,
    'Avg. Ticket Price To Date': avgTicketPrice
  };

  return <ChartTooltip headerText={headerText} bodyJson={bodyJson} />;
}
