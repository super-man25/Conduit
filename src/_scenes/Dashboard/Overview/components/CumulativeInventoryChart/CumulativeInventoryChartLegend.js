import React, { Fragment } from 'react';
import { cssConstants } from '_constants';
import { ChartLegendItem } from '_components';

export function CumulativeInventoryChartLegend({ hasProjected }) {
  return (
    <Fragment>
      <ChartLegendItem
        label={hasProjected ? 'Actual Revenue' : 'Revenue'}
        color={cssConstants.SECONDARY_PURPLE}
      />
      {hasProjected && (
        <ChartLegendItem
          dashed
          color={cssConstants.SECONDARY_LIGHT_PURPLE}
          label="Projected Inventory"
        />
      )}
    </Fragment>
  );
}
