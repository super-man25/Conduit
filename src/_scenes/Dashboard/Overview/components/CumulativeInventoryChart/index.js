export * from './CumulativeInventoryChart';
export * from './CumulativeInventoryTooltip';
export * from './CumulativeInventoryChartLegend';
