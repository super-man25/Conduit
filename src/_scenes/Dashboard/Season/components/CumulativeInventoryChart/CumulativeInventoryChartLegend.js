import React, { Fragment } from 'react';
import { cssConstants } from '_constants';
import { ChartLegendItem } from '_components';

export function CumulativeInventoryChartLegend() {
  return (
    <Fragment>
      <ChartLegendItem
        label="Actual Inventory"
        color={cssConstants.SECONDARY_LIGHT_PURPLE}
      />
      <ChartLegendItem
        dashed
        color={cssConstants.SECONDARY_LIGHT_PURPLE}
        label="Projected Inventory"
      />
    </Fragment>
  );
}
