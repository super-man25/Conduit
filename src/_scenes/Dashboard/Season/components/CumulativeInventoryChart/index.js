export * from './CumulativeInventoryChart';
export * from './CumulativeInventoryTooltip';
