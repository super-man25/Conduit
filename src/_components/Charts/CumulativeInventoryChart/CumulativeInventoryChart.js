// @flow
import * as React from 'react';
import {
  ResponsiveContainer,
  LineChart,
  Line,
  XAxis,
  YAxis,
  CartesianGrid,
  Label,
  Tooltip
} from 'recharts';
import { cssConstants, chartLabelStyles } from '_constants';
import { ChartContainer } from '_components';
import { CumulativeInventoryTooltip } from './CumulativeInventoryTooltip';
import { truncateNumber } from '_helpers/string-utils';
import type { EventStat } from '_models/eventStat';
import { isMobileDevice } from '_helpers';

const ANIMATION_DURATION = 1000;

type Props = {
  height: number,
  data: EventStat[],
  totalInventory: number,
  tooltipDateFormatter: (Date) => string,
  renderNoData: () => React.Node,
  majorXAxisTickRenderer: () => HTMLElement,
  minorXAxisTickRenderer: () => HTMLElement,
  mobileXAxisTickRenderer: () => HTMLElement
};

export function CumulativeInventoryChart({
  height,
  data,
  totalInventory,
  tooltipDateFormatter,
  renderNoData,
  majorXAxisTickRenderer,
  minorXAxisTickRenderer,
  mobileXAxisTickRenderer
}: Props) {
  if (!data.length) {
    return renderNoData();
  }

  return (
    <ChartContainer>
      <ResponsiveContainer width="100%" height={height}>
        <LineChart data={data}>
          {isMobileDevice ? (
            <XAxis
              dataKey="timestamp"
              tick={mobileXAxisTickRenderer}
              axisLine={false}
              tickLine={false}
              interval={0}
            />
          ) : (
            <>
              <XAxis
                dataKey="timestamp"
                tick={minorXAxisTickRenderer}
                tickLine={false}
                interval={0}
                height={30}
              />
              <XAxis
                dataKey="timestamp"
                tick={majorXAxisTickRenderer}
                xAxisId="majorXAxis"
                axisLine={false}
                tickLine={false}
                interval={0}
                height={10}
              />
            </>
          )}

          <YAxis
            tickCount={10}
            tickFormatter={truncateNumber}
            tickLine={false}
            axisLine={false}
            tick={{ fontSize: 10 }}
            domain={['auto', 'auto']}
          >
            <Label
              value="Inventory"
              angle={-90}
              position="insideLeft"
              style={chartLabelStyles}
            />
          </YAxis>

          <CartesianGrid vertical={false} />

          <Line
            animationDuration={ANIMATION_DURATION}
            type="monotone"
            dataKey="inventory"
            name="actual"
            dot={false}
            strokeWidth={2}
            stroke={cssConstants.SECONDARY_PURPLE}
          />

          <Line
            animationDuration={ANIMATION_DURATION}
            type="monotone"
            dataKey="projectedInventory"
            strokeDasharray="5 5"
            dot={false}
            name="projected"
            strokeWidth={2}
            stroke={cssConstants.SECONDARY_LIGHT_PURPLE}
          />
          <Tooltip
            cursor={{
              stroke: cssConstants.SECONDARY_LIGHT_PURPLE,
              strokeWidth: 2,
              opacity: 0.5
            }}
            content={
              <CumulativeInventoryTooltip
                dateFormatter={tooltipDateFormatter}
                totalInventory={totalInventory}
              />
            }
          />
        </LineChart>
      </ResponsiveContainer>
    </ChartContainer>
  );
}
