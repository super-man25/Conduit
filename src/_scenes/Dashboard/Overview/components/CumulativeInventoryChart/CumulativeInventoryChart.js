// @flow
import * as React from 'react';
import {
  ResponsiveContainer,
  LineChart,
  Line,
  XAxis,
  YAxis,
  CartesianGrid,
  Label,
  Tooltip,
} from 'recharts';

import { chartLabelStyles, colors } from '_constants';
import { ChartContainer } from '_components';
import { CumulativeInventoryTooltip } from './CumulativeInventoryTooltip';
import { truncateNumber } from '_helpers/string-utils';
import type { EventStat } from '_models/eventStat';
import { isMobileDevice } from '_helpers';

const animationDuration = 1000;

type Props = {
  height: number,
  data: EventStat[],
  totalInventory: number,
  tooltipDateFormatter: (Date) => string,
  renderNoData: () => React.Node,
  majorXAxisTickRenderer: () => HTMLElement,
  minorXAxisTickRenderer: () => HTMLElement,
  mobileXAxisTickRenderer: () => HTMLElement,
};

export function CumulativeInventoryChart({
  height,
  data,
  totalInventory,
  tooltipDateFormatter,
  renderNoData,
  majorXAxisTickRenderer,
  minorXAxisTickRenderer,
  mobileXAxisTickRenderer,
}: Props) {
  if (!data.length) {
    return renderNoData();
  }

  return (
    <ChartContainer>
      <ResponsiveContainer width="100%" height={height}>
        <LineChart data={data}>
          {isMobileDevice && (
            <XAxis
              dataKey="timestamp"
              tick={mobileXAxisTickRenderer}
              axisLine={false}
              tickLine={false}
              interval={0}
            />
          )}
          {!isMobileDevice && (
            <XAxis
              dataKey="timestamp"
              tick={minorXAxisTickRenderer}
              tickLine={false}
              interval={0}
              height={30}
            />
          )}
          {!isMobileDevice && (
            <XAxis
              dataKey="timestamp"
              tick={majorXAxisTickRenderer}
              xAxisId="majorXAxis"
              axisLine={false}
              tickLine={false}
              interval={0}
              height={10}
            />
          )}

          <YAxis
            tickCount={10}
            tickFormatter={truncateNumber}
            tickLine={false}
            axisLine={false}
            tick={{ fontSize: 10 }}
            domain={['auto', 'auto']}
          >
            <Label
              value="Inventory"
              angle={-90}
              position="insideLeft"
              style={chartLabelStyles}
            />
          </YAxis>

          <CartesianGrid vertical={false} />

          <Line
            animationDuration={animationDuration}
            type="monotone"
            dataKey="inventory"
            name="actual"
            dot={false}
            strokeWidth={2}
            stroke={colors.blue}
          />

          <Line
            animationDuration={animationDuration}
            type="monotone"
            dataKey="projectedInventory"
            strokeDasharray="5 5"
            dot={false}
            name="projected"
            strokeWidth={2}
            stroke={colors.neonBlue}
          />
          <Tooltip
            cursor={{
              stroke: colors.gray,
              strokeWidth: 2,
              opacity: 0.25,
            }}
            content={
              <CumulativeInventoryTooltip
                dateFormatter={tooltipDateFormatter}
                totalInventory={totalInventory}
              />
            }
          />
        </LineChart>
      </ResponsiveContainer>
    </ChartContainer>
  );
}
