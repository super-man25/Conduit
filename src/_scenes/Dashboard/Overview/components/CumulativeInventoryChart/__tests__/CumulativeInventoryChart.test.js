import React from 'react';
import { shallow } from 'enzyme';
import { CumulativeInventoryChart } from '../CumulativeInventoryChart';

const createProps = () => ({
  height: 400,
  data: [
    {
      eventId: 1,
      id: 1,
      inventory: 100,
      periodicInventory: 100,
      revenue: 100,
      periodicRevenue: 100,
      isProjected: false,
      timestamp: 1529336865706,
    },
  ],
  dateFormatter: (d) => d.toString,
  renderNoData: () => <div />,
});

describe('<CumulativeInventoryChart />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<CumulativeInventoryChart {...createProps()} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly with no date', () => {
    const props = { ...createProps(), data: [] };
    const wrapper = shallow(<CumulativeInventoryChart {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly with a different height', () => {
    const props = { ...createProps(), height: 300 };
    const wrapper = shallow(<CumulativeInventoryChart {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
