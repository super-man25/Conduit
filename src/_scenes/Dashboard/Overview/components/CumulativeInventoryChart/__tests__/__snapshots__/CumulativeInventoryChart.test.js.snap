// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CumulativeInventoryChart /> should render correctly 1`] = `ShallowWrapper {}`;

exports[`<CumulativeInventoryChart /> should render correctly with a different height 1`] = `ShallowWrapper {}`;

exports[`<CumulativeInventoryChart /> should render correctly with no date 1`] = `ShallowWrapper {}`;
